ros_cross_compile/builders.py

```diff
@@ -15,27 +15,27 @@
 import os
 from pathlib import Path
 
-import docker
+from ros_cross_compile.docker_client import DockerClient
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
-def run_emulated_docker_build(image_tag: str, workspace_path: Path) -> None:
+def run_emulated_docker_build(
+    docker_client: DockerClient, image_tag: str, workspace_path: Path
```

Comment:
Do we need to worry about docker_client storing some state or is it just executing Docker commands? My concern is whether docker_client will or should be immutable or not down the line.

Reply:
it's a valid concern, however there is no mutable state exposed by the client. if you look at my latest pushed commit - always passing the client makes the code more testable by allowing mocking.